packages/io-messages-common/src/adapters/lollipop/__tests__/lollipop-client.test.ts

```
@@ -0,0 +1,104 @@
+
+    const lcParams = await client.generateLCParams(
+      anAssertionRef,
+      aSignatureInput,
+    );
+
+    expect(lcParams).toEqual(anLcParams);
+    expect(fetchSpy).toHaveBeenCalledWith(
+      `${baseUrl}/pubKeys/${anAssertionRef}/generate`,
+      expect.objectContaining({
+        body: JSON.stringify({ operationId: aSignatureInput }),
+        headers: expect.objectContaining({
+          "X-Functions-Key": apiKey,
+          "content-type": "application/json",
+        }),
+        method: "POST",
+      }),
+    );
+  });
+
+  it("throws an error when the API responds with an error", async () => {
+    const fetchSpy = vi.spyOn(globalThis, "fetch").mockResolvedValueOnce({
+      json: async () => aProblemJson,
+      ok: false,
+      status: 400,
+    } as Response);
+
+    try {
+      await client.generateLCParams(anAssertionRef, aSignatureInput);
+      throw new Error("Expected error to be thrown");
+    } catch (error) {
+      const typedError = error as Error;
+      expect(typedError.message).toBe(
+        `Error during generateLcParams with status ${aProblemJson.status} and body ${JSON.stringify(aProblemJson)}`,
+      );
+    }
+
+    expect(fetchSpy).toHaveBeenCalledWith(
+      `${baseUrl}/pubKeys/${anAssertionRef}/generate`,
+      expect.objectContaining({
+        body: JSON.stringify({ operationId: aSignatureInput }),
+        headers: expect.objectContaining({
+          "X-Functions-Key": apiKey,
+          "content-type": "application/json",
+        }),
+        method: "POST",
+      }),
+    );
+
+    expect(fetchSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it("should throw an error if fetch fails", async () => {
+    const returnedError = new Error("Network error");
+    const fetchSpy = vi
+      .spyOn(globalThis, "fetch")
+      .mockRejectedValueOnce(returnedError);
+
+    try {
+      await client.generateLCParams(anAssertionRef, aSignatureInput);
+      throw new Error("Expected error to be thrown");
```

Why this throw here? the generateLCParams should throw the error
```suggestion
```

---

[here](https://github.com/pagopa/io-messages/pull/410/commits/55fc0fbd179379bd886e224feee8a2642ce61ae8)